kolibri/core/content/management/commands/importchannel.py

```diff
@@ -174,12 +175,16 @@ def _start_file_transfer(  # noqa: C901
                                 .exclude(kind=content_kinds.TOPIC)
                                 .values_list("id", flat=True)
                             )
-                            import_ran = import_channel_by_id(
-                                channel_id, self.is_cancelled
-                            )
+                            with db_task_write_lock:
```

Comment:
any reason not to group the two statements into a single lock?

Reply:
We could, but I was trying to make it more granular to prevent any task hogging the DB for too long.